tableauserverclient/models/schedule_item.py

```diff
@@ -0,0 +1,203 @@
+            (_, name, _, _, updated_at, _, frequency, next_run_at, end_schedule_at, execution_order,
+             priority, interval_item) = self._parse_element(schedule_xml)
+
+            self._set_values(None, name, None, None, updated_at, None, frequency, next_run_at, end_schedule_at,
+                             execution_order, priority, interval_item)
+
+        return self
+
+    def _set_values(self, id, name, state, created_at, updated_at, schedule_type, frequency,
+                    next_run_at, end_schedule_at, execution_order, priority, interval_item):
+        if id is not None:
+            self._id = id
+        if name:
+            self._name = name
+        if state:
+            self._state = state
+        if created_at:
+            self._created_at = created_at
+        if updated_at:
+            self._updated_at = updated_at
+        if schedule_type:
+            self._schedule_type = schedule_type
+        if frequency:
+            self._frequency = frequency
+        if next_run_at:
+            self._next_run_at = next_run_at
+        if end_schedule_at:
+            self._end_schedule_at = end_schedule_at
+        if execution_order:
+            self._execution_order = execution_order
+        if priority:
+            self._priority = priority
+        if interval_item:
+            self._interval_item = interval_item
+
+    @classmethod
+    def from_response(cls, resp):
+        all_schedule_items = list()
+        parsed_response = ET.fromstring(resp)
+        all_schedule_xml = parsed_response.findall('.//t:schedule', namespaces=NAMESPACE)
+        for schedule_xml in all_schedule_xml:
+            (id, name, state, created_at, updated_at, schedule_type, frequency, next_run_at,
+             end_schedule_at, execution_order, priority, interval_item) = cls._parse_element(schedule_xml)
+
+            schedule_item = cls(name, priority, schedule_type, execution_order, interval_item)
+            schedule_item._set_values(id, None, state, created_at, updated_at, None, frequency, next_run_at,
+                                      end_schedule_at, None, None, None)
+            all_schedule_items.append(schedule_item)
+        return all_schedule_items
+
+    @staticmethod
+    def _parse_element(schedule_xml):
+        id = schedule_xml.get('id', None)
+        name = schedule_xml.get('name', None)
+        state = schedule_xml.get('state', None)
+        created_at = schedule_xml.get('createdAt', None)
+        updated_at = schedule_xml.get('updatedAt', None)
+        schedule_type = schedule_xml.get('type', None)
+        frequency = schedule_xml.get('frequency', None)
+        next_run_at = schedule_xml.get('nextRunAt', None)
+        end_schedule_at = schedule_xml.get('endScheduleAt', None)
+        execution_order = schedule_xml.get('executionOrder', None)
+
+        priority = schedule_xml.get('priority', None)
+        if priority:
+            priority = int(priority)
```

Comment:
This will fail if priority is a non empty string that isn't an integer. Do we know that's always safe to do this conversion?